src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/UnixIPGlobalProperties.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net.NetworkInformation
+{
+    internal abstract class UnixIPGlobalProperties : IPGlobalProperties
+    {
+        public sealed override Task<UnicastIPAddressInformationCollection> GetUnicastAddressesAsync()
+        {
+            return Task.Run((Func<UnicastIPAddressInformationCollection>)GetUnicastAddresses);
```

Comment:
It'd be slightly more efficient to do:
```C#
return Task.Factory.StartNew(s => ((UnixIPGobalProperties)s).GetUnicastAddresses(), this,
    CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
```
That'll avoid needing to allocate a new delegate each time this is called.